eng/release/DiagnosticsReleaseTool/Common/AzureBlobPublisher.cs

```diff
@@ -0,0 +1,267 @@
+            }
+        }
+
+        private StorageSharedKeyCredential AccountCredential
+        {
+            get
+            {
+                StorageSharedKeyCredential credential = new StorageSharedKeyCredential(_accountName, _accountKey);
+                return credential;
+            }
+        }
+
+        private BlobClientOptions BlobOptions
+        {
+            get
+            {
+                // The Azure SDK client has it's own built in retry logic
+                // We want to allow more and longer retries because this
+                // is a publishing operation that happens once and can be
+                // allowed to take a very long time. We have a high
+                // tolerance for slow operations and a low tolerance for failure.
+                return new BlobClientOptions()
+                {
+                    Retry =
+                    {
+                        MaxRetries = MaxRetries,
+                    }
+                };
+            }
+        }
+
+        public AzureBlobBublisher(string accountName, string accountKey, string containerName, string releaseName, int sasValidDays, ILogger logger)
+        {
+            _accountName = accountName;
+            _accountKey = accountKey;
+            _containerName = containerName;
+            _releaseName = releaseName;
+            _sasValidDays = sasValidDays;
+            _logger = logger;
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public async Task<string> PublishFileAsync(FileMapping fileMap, CancellationToken ct)
+        {
+            Uri result = null;
+            int retriesLeft = MaxFullLoopRetries;
+            TimeSpan loopDelay = FullLoopRetryDelay;
+            bool completed = false;
+
+            do
+            {
+                _logger.LogInformation($"Attempting to publish {fileMap.RelativeOutputPath}, {retriesLeft} tries left.");
+                try
+                {
+                    BlobContainerClient client = await GetClient(ct);
+                    if (client == null)
+                    {
+                        // client creation failed, return
+                        return null;
+                    }
+
+                    using var srcStream = new FileStream(fileMap.LocalSourcePath, FileMode.Open, FileAccess.Read);
+
+                    BlobClient blobClient = client.GetBlobClient(GetBlobName(_releaseName, fileMap.RelativeOutputPath));
+
+                    await blobClient.UploadAsync(srcStream, overwrite: true, ct);
+
+                    BlobSasBuilder sasBuilder = new BlobSasBuilder()
+                    {
+                        BlobContainerName = client.Name,
+                        BlobName = blobClient.Name,
+                        Identifier = AccessPolicyDownloadId,
+                        Protocol = SasProtocol.Https,
+                    };
+                    Uri accessUri = blobClient.GenerateSasUri(sasBuilder);
```

Comment:
Did you want to be explicit on the expiration here?

Reply:
It's settable explicitly in the pipeline.